ext/js/language/language-util.js

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2024  Yomitan Authors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+/** @type {import('language').TextPreprocessorOptions<boolean>} */
+export const basicTextPreprocessorOptions = [false, true];
+
+/** @type {import('language-util').CodepointRange} */
+const CJK_UNIFIED_IDEOGRAPHS_RANGE = [0x4e00, 0x9fff];
```

Comment:
I think there's a few approaches we could take with organization here:
1. Arabic and Farsi share the arabic script, but I chose to leave removaArabicDiacritics in arabic since I guess it's the 'main' language for that script. The Han char stuff could stay on Japanese or be moved to Chinese.
2. Make files like 'han-characters.js', 'arabic-script.js', 'latin-script.js' to keep things common to a few languages.
3. Put common stuff in the 'globally' common files - `language-util.js`,  `text-preprocessors.js` etc. I did this with the capitalization thinking that it is ubiquitous enough. 

Here, I'd go with `han-characters.js`.

Reply:
Still needs a common file to keep things like `isCodePointInRange`.